kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/api/AdapterGenerator.kt

```diff
@@ -198,10 +197,18 @@ internal class AdapterGenerator(
       }
     }
 
-    val maskName = nameAllocator.newName("mask")
+    // Calculate how many masks we'll need. Round up if it's not evenly divisible by 32
+    val maskCount = ceil(propertyList.count { it.hasConstructorParameter }.toFloat().div(32)).toInt()
```

Comment:
What's with the weird math? You should never need floats to do int math.

Conventional int math looks like this:
val maskCount = (propertyList.count { it.hasConstructorParameter } + 31) / 32

Reply:
Also what's the behavior on zero parameters? 0 masks or 1?

Reply:
I was just rounding up, that approach works too. Good point on 0 parameters, will add a check for that.

Reply:
497fdba for the math
7f4532e for the empty check. Worth mentioning that even before, it would just ignore the mask names in that case since it wouldn't use the dynamic invocation in that case